Sources/SpeziLLMLocal/LLMLocalPlatform.swift

```diff
@@ -58,18 +60,26 @@ public actor LLMLocalPlatform: LLMPlatform, DefaultInitializable {
     
     public nonisolated func configure() {
 #if targetEnvironment(simulator)
-        assertionFailure("SpeziLLMLocal: Code cannot be run on simulator.")
-#endif
+        Logger(
+            subsystem: "Spezi",
+            category: "LLMLocalPlatform"
+        ).warning("SpeziLLMLocal is only supported on physical devices. Use `LLMMockPlatform` instead.")
+#else
         if let cacheLimit = configuration.cacheLimit {
             MLX.GPU.set(cacheLimit: cacheLimit * 1024 * 1024)
         }
         if let memoryLimit = configuration.memoryLimit {
             MLX.GPU.set(memoryLimit: memoryLimit.limit, relaxed: memoryLimit.relaxed)
         }
+#endif
     }
     
-    public nonisolated func callAsFunction(with llmSchema: LLMLocalSchema) -> LLMLocalSession {
+    public nonisolated func callAsFunction(with llmSchema: LLMLocalSchema) -> some LLMSession {
+        #if targetEnvironment(simulator)
```

Comment:
Yes, I’m not entirely satisfied with this solution either. To address the issue, we now display an additional warning in the mock context (output): `Using SpeziLLMLocal only works on physical devices.`.

This is why I created a separate `LLMLocalMockSession` instead of reusing `LLMMockSession`, which would have required changes to the `LLMPlatform`.

Without this extra mock session, the simulator is unsupported, causing the app to crash when the platform is initialized.

If you have any suggestions or recommendations, I’d be more than happy to explore them and implement a better solution! 🚀

Reply:
Thanks for elaborating! I'm fine with the solution as is, I just really recommend that you document all of that really well in README, DocC, and in-line.